### PR TITLE
Submit reviews responder

### PR DESCRIPTION
**Pull request by xuanxu**
This PR adds a responder listening to the following command: `submit review <REVIEW-URL> time <REVIEW-HOURS>`
The review URL, duration and date are logged in Airtable. 
The responder can also add or remove labels when the number of reviews equals the number of reviewers.

Re: #9

Here's a screenshot of this responder's docs:
ROpenSci Submit review

### CODE REVIEW

**Comment by maelle** [comment]
@xuanxu Thanks a ton!

Will it work if people mistakenly write "3.5h", "3.5 h", "1hour", "3 hours" etc?

**Comment by maelle** [comment]
> Note: currently the review_date column in the reviews table is only present in the test Airtable database, it is missing from the production database.

Thanks, now added to the production database.

**Comment by maelle** [comment]
Also, does the "all reviews" completed depend on the reviewers assigned? (as opposed to an arbitrary number of reviews)

**Comment by mpadge** [comment]
> Also, does the "all reviews" completed depend on the reviewers assigned? (as opposed to an arbitrary number of reviews)

That is critically important, because there definitely are, and will be, cases with `num_reviewers != 2`.

**Comment by xuanxu** [comment]
> Will it work if people mistakenly write "3.5h", "3.5 h", "1hour", "3 hours" etc?

Currently it would only accept "3", "3.5", "3,5", with no text after that: "submit review URW time 6.5". 
Should I change it to accept also the [h/hour/hours] part?

**Comment by xuanxu** [comment]
> Also, does the "all reviews" completed depend on the reviewers assigned?

Yes, the responder queries the airtable API for the reviews of all the assigned reviewers, and if they are all complete (review_url is not empty), then it consider all reviews are in and add the label. It works with any number of reviewers in the list. In the example in the docs there's only two, but could be any other number of reviewers.

**Comment by maelle** [comment]
Thanks @xuanxu!

> Should I change it to accept also the [h/hour/hours] part?

Yes I think it'd be safer (with or without the space before the h).

**Comment by xuanxu** [comment]
>> Should I change it to accept also the [h/hour/hours] part?
>
> Yes I think it'd be safer (with or without the space before the h).

@maelle Done!

**Review by mpadge — CHANGES_REQUESTED** [review]
@xuanxu Is it also possible to add an error at the start of `process_message` if `review_time` is in format `HH:MM`? That may happen, and it should be easy to `regex` that pattern and divert to a `respond("Error: ..."); return` clause, right?

**Comment by xuanxu** [comment]
@mpadge right now that format is not accepted and the command will be ignored. Once we have an "I dont understand that" responder, this will happen automatically. Do still want a specific error message for this case?

**Comment by mpadge** [comment]
Only  because i though that would be easy to do, and then useful to have a specific response telling people that their time values aren't properly formatted. Your call, but if it's not too difficult, then yes, please!

**Comment by maelle** [comment]
I'll wait for @xuanxu's answer before merging and testing this.

**Comment by xuanxu** [comment]
I've added a simple validation: if the time value includes a colon `:` the response will be an "Invalid time format" comment.

**Comment by maelle** [comment]
Thank you @xuanxu!!

File: app/responders/ropensci/submit_review_responder.rb
Comment (mpadge) [inline app/responders/ropensci/submit_review_responder.rb:18]: Great @xuanxu, that's exactly what i was imaging! I think that will be very useful - thank you

**Pull request merged**
